scripts/solutions.sh

```diff
@@ -0,0 +1,492 @@
+            ;;
+        -N|--namespace)
+            shift
+            NAMESPACE=$1
+            if ! check_namespace_validity "$NAMESPACE"; then
+                echo 1>&2 "Namespace name '$NAMESPACE' is invalid: it must" \
+                    "consist of lower case alphanumeric characters or '-'," \
+                    "and must start and end with an alphanumeric character."
+                exit 1
+            fi
+            ;;
+        -s|--solution)
+            shift
+            SOLUTION=$1
+            ;;
+        -v|--verbose)
+            VERBOSE=1
+            ;;
+        -V|--version)
+            shift
+            VERSION=$1
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            echo 1>&2 "Option parsing failure"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+TMPFILES=$(mktemp -d)
+
+mkdir -p "$(dirname "$LOGFILE")"
+
+exec > >(tee -ia "$LOGFILE") 2>&1
+
+cleanup() {
+    rm -rf "$TMPFILES" || true
+}
+
+trap cleanup EXIT
+
+BASE_DIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
+# shellcheck disable=SC1090
+. "$BASE_DIR"/common.sh
+
+check_command_mandatory_options() {
+    local -a missing_options=()
+    local -r command=$1
+
+    for option in ${COMMAND_MANDATORY_OPTIONS[$command]:-}; do
+        [[ ${!OPTIONS_MAPPING[$option]:-} ]] || missing_options+=("$option")
+    done
+
+    if (( ${#missing_options[@]} )); then
+        echo 1>&2 "Missing options for command '$command':" \
+                  "${missing_options[@]}"
+        return 1
+    fi
+
+    return 0
+}
+
+salt_minion_exec() {
+    salt-call "$@" --retcode-passthrough
+}
+
+salt_master_exec() {
+    SALTENV=${SALTENV:-$(get_salt_env)}
+    SALT_MASTER_CONTAINER_ID=${SALT_MASTER_CONTAINER_ID:-$(get_salt_container)}
+    crictl exec -i "$SALT_MASTER_CONTAINER_ID" "$@" saltenv="$SALTENV"
+}
+
+activate_solution() {
+    run "Updating Solutions configuration file" \
+        salt_minion_exec metalk8s_solutions.activate_solution \
+        solution="$NAME" \
+        version="$VERSION" \
+        --local
+
+    run "Deploying Solution components" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
+}
+
+deactivate_solution() {
+    run "Updating Solutions configuration file" \
+        salt_minion_exec metalk8s_solutions.deactivate_solution \
+        solution="$NAME" \
+        --local
+
+    run "Removing Solution components" \
+        salt_master_exec salt-run state.orchestrate \
+        metalk8s.orchestrate.solutions.deploy-components \
+        pillar="{'bootstrap_id': '$(get_salt_minion_id)'}"
+}
+
+configure_archives() {
+    local removed=${1:-False}
+
+    for archive in "${ARCHIVES[@]}"; do
+        salt_minion_exec metalk8s_solutions.configure_archive \
+            archive="$archive" \
+            removed="$removed" \
+            --local
```

Comment:
You should set `create_config=True` because here that will fail 
```
Error running 'metalk8s_solutions.configure_archive': Failed to load "/etc/metalk8s/solutions.yaml": [Errno 2] No such file or directory: '/etc/metalk8s/solutions.yaml'
```

Also here you have a bash issue as this fail silently here and the `refresh_pillar` return 0 so the next state is executed also (I through that was already fixed in another PR :/ )
```
Failure while running step 'Importing Solutions'

Command: salt_minion_exec state.sls metalk8s.solutions.available saltenv=metalk8s-2.4.3-dev

Output:

<< BEGIN >>
[CRITICAL] Rendering SLS 'metalk8s-2.4.3-dev:metalk8s.solutions.available' failed: while parsing a block mapping
  in "<unicode string>", line 11, column 7:
        - comment: "Errors: Error when rea ... 
          ^
expected <block end>, but found '<scalar>'
  in "<unicode string>", line 11, column 83:
     ... ns config file: Failed to load "/etc/metalk8s/solutions.yaml": [ ... 
                                         ^
local:
    Data failed to compile:
----------
    Rendering SLS 'metalk8s-2.4.3-dev:metalk8s.solutions.available' failed: while parsing a block mapping
  in "<unicode string>", line 11, column 7:
        - comment: "Errors: Error when rea ... 
          ^
expected <block end>, but found '<scalar>'
  in "<unicode string>", line 11, column 83:
     ... ns config file: Failed to load "/etc/metalk8s/solutions.yaml": [ ... 
                                         ^
<< END >>
```

Reply:
For the first point, I assumed that the user have to create this file first, maybe I'm wrong.

Reply:
done